<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RuleEdit</class>
 <widget class="QWidget" name="RuleEdit">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>581</width>
    <height>611</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="margin">
    <number>9</number>
   </property>
   <item>
    <widget class="QGroupBox" name="generalGroup">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>General</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::FieldsStayAtSizeHint</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="nameLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Rule name:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="nameEdit"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="documentationLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Documentation:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPlainTextEdit" name="documentationEdit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>100</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Left-hand side</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Developer::GraphWidget" name="lhsGraph"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Right-hand side</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Developer::GraphWidget" name="rhsGraph"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="variablesLayout">
       <item>
        <layout class="QHBoxLayout" name="variablesLabelLayout">
         <item>
          <widget class="QLabel" name="variablesLabel">
           <property name="text">
            <string>Variables</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableWidget" name="variablesWidget">
         <property name="maximumSize">
          <size>
           <width>260</width>
           <height>16777215</height>
          </size>
         </property>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <row>
          <property name="text">
           <string>New Row</string>
          </property>
         </row>
         <row>
          <property name="text">
           <string>New Row</string>
          </property>
         </row>
         <row>
          <property name="text">
           <string>New Row</string>
          </property>
         </row>
         <column>
          <property name="text">
           <string>Identifier</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Type</string>
          </property>
         </column>
         <item row="0" column="0">
          <property name="text">
           <string/>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="behaviourGroup">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Behavior</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="matchRulesLayout">
        <item>
         <widget class="QLabel" name="matchRulesLabel1">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Graph Matching should be:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="matchRulesCombo">
          <item>
           <property name="text">
            <string>Injective</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Non-injective</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="icons.qrc">
            <normaloff>:/icons/small_help.png</normaloff>:/icons/small_help.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="conditionsLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Only match when the following  condition(s) evaluate to true:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Developer::ConditionEditor" name="conditionsEdit"/>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Developer::GraphWidget</class>
   <extends>QGraphicsView</extends>
   <header>graphview/graphwidget.hpp</header>
  </customwidget>
  <customwidget>
   <class>Developer::ConditionEditor</class>
   <extends>QPlainTextEdit</extends>
   <header>conditioneditor.hpp</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>pressed()</signal>
   <receiver>RuleEdit</receiver>
   <slot>showInjectiveHelp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>408</x>
     <y>358</y>
    </hint>
    <hint type="destinationlabel">
     <x>580</x>
     <y>384</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nameEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>RuleEdit</receiver>
   <slot>nameChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>318</x>
     <y>47</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>documentationEdit</sender>
   <signal>textChanged()</signal>
   <receiver>RuleEdit</receiver>
   <slot>documentationChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>449</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>620</x>
     <y>119</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>matchRulesCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>RuleEdit</receiver>
   <slot>injectiveChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>312</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel">
     <x>612</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>conditionsEdit</sender>
   <signal>textChanged()</signal>
   <receiver>RuleEdit</receiver>
   <slot>conditionChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>508</x>
     <y>535</y>
    </hint>
    <hint type="destinationlabel">
     <x>607</x>
     <y>599</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>showInjectiveHelp()</slot>
  <slot>nameChanged(QString)</slot>
  <slot>documentationChanged()</slot>
  <slot>lhsChanged()</slot>
  <slot>rhsChanged()</slot>
  <slot>injectiveChanged(int)</slot>
  <slot>conditionChanged()</slot>
 </slots>
</ui>
